### PR TITLE
Ret 4732 - Add notifications extract

### PR DESCRIPTION
**Pull request by marcjudge**
### Jira link (if applicable)
https://tools.hmcts.net/jira/browse/RET-4732

### Change description ###
Add notifications extract

### Checklist

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)
- [ ] Does this PR introduce a breaking change

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/multiples/ExtractNotificationsController.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        @ApiResponse(responseCode = "500", description = "Internal Server Error")
+    })
+    public ResponseEntity<MultipleCallbackResponse> aboutToSubmit(@RequestBody MultipleRequest multipleRequest,
+                                                                  @RequestHeader("Authorization") String userToken)
+            throws IOException {
+
+        List<String> errors = new ArrayList<>();
+        MultipleDetails multipleDetails = multipleRequest.getCaseDetails();
+
+        log.info("Generating report");
```
[the remark this answers is not in the document]
Comment (marcjudge) [reply src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/multiples/ExtractNotificationsController.java]: Updated

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/multiples/MultipleNotificationsHelper.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+import static java.util.stream.Collectors.groupingBy;
+import static org.apache.commons.collections.CollectionUtils.isEmpty;
+import static uk.gov.hmcts.ecm.common.model.helper.Constants.NO;
+import static uk.gov.hmcts.ecm.common.model.helper.Constants.YES;
+
+@Slf4j
+public final class MultipleNotificationsHelper {
+
+    public static final String DATE_FORMAT = "dd MMM yyyy";
```
[the remark this answers is not in the document]
Comment (marcjudge) [reply src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/multiples/MultipleNotificationsHelper.java]: Might as well update everywhere :) done

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/excel/SingleCasesReadingService.java [hunk 3]
```diff
@@ -104,4 +105,19 @@ public Set<SchedulePayloadEvent> retrieveScheduleCases(String userToken, String
...
+                                                                                   List<String> caseIds) {
+        HashSet<NotificationSchedulePayloadEvent> schedulePayloadEvents = new HashSet<>();
+        try {
+            schedulePayloadEvents = new HashSet<>(ccdClient.retrieveCasesElasticNotificationSearchSchedule(userToken,
+                    UtilHelper.getCaseTypeId(multipleCaseTypeId),
+                    caseIds));
+
+        } catch (Exception ex) {
+            log.error("Error retrieving notification schedule cases");
+            log.error(ex.getMessage(), ex);
```
[the remark this answers is not in the document]
Comment (marcjudge) [reply src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/excel/SingleCasesReadingService.java]: Done

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/multiples/ExtractNotificationsController.java [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+    private final NotificationsExcelReportService notificationsExcelReportService;
+
+    /**
+     * Creates the Excel report.
+     *
+     * @param multipleRequest holds the request and case data
+     * @return Callback response entity with case data attached.
+     */
+    @PostMapping(value = "/aboutToSubmit", consumes = APPLICATION_JSON_VALUE)
+    @Operation(summary = "submitted")
```
[the remark this answers is not in the document]
Comment (marcjudge) [reply src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/multiples/ExtractNotificationsController.java:42]: Made it more descriptive now

[the remark this answers is not in the document]
Comment (hsjhita) [reply src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/multiples/ExtractNotificationsController.java:42]: this was more about the value of the operation annotation

**Review by hsjhita — DISMISSED** [review]

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/multiples/ExtractNotificationsController.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (marcjudge) [reply src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/multiples/ExtractNotificationsController.java]: Ah sorry, fixed

**Review by hsjhita — APPROVED** [review]

**Review by jip1984 — APPROVED** [review]

**Pull request merged**